docs/make.jl

```diff
@@ -2,6 +2,15 @@ using Documenter
 using HDF5
 using MPI  # needed to generate docs for parallel HDF5 API
 
+# Used in index.md to filter the autodocs list
+wherefrom(m::Method) = String(m.file)
+not_from_api(m::Method) = !endswith(wherefrom(m), "src/api.jl")
```

Comment:
mabye not worth to add the `wherefrom` method 
```suggestion
not_from_api(m::Method) = !endswith(wherefrom(String(m.file)), "src/api.jl")
```